RS/2D vector space/main.py

```diff
@@ -0,0 +1,137 @@
+        if event.type == QUIT:
+            pygame.quit()
+            sys.exit()
+
+    pxArray = pygame.PixelArray(picture)
+
+    for x in range(0, windowWidth - 1):
+        for y in range(0, windowHeight - 1):
+            currentRed = window.get_at((x, y)).r
+            currentGreen = window.get_at((x, y)).g
+            currentBlue = window.get_at((x, y)).b
+            currentAlpha = window.get_at((x, y)).a
+
+            pxArray[currentRed, currentGreen] = RED
+            pxArray[currentRed, currentBlue] = BLACK
+            pxArray[currentGreen, currentRed] = GREEN
+            pxArray[currentGreen, currentBlue] = BLACK
+            pxArray[currentBlue, currentRed] = BLUE
+            pxArray[currentBlue, currentGreen] = BLACK
+
+    print ('checkpoint2')
+
+    pict = pxArray.make_surface()
+
+    del pxArray
+
+    print('checkpoint3')
+
+    window.blit(pict, (0, 0))
+
+    pygame.display.update()
+
+    print('checkpoint4')
+
+    firstPart = False
+    secondPart = True
+
+
+while secondPart == True:
+
+    for event in pygame.event.get():
+        if event.type == QUIT:
+            pygame.quit()
+            sys.exit()
+
+    pxArray = pygame.PixelArray(picture)
+
+    for x in range(0, windowWidth - 1):
+        for y in range(0, windowHeight - 1):
+            currentRed = window.get_at((x, y)).r
+            currentGreen = window.get_at((x, y)).g
+            currentBlue = window.get_at((x, y)).b
+
+            luminosity = math.sqrt(math.pow(currentRed, 2) +
+                                   math.pow(currentGreen, 2) +
+                                   math.pow(currentBlue, 2))
+            if luminosity != 0 or luminosity < 440:
+                lumTotal += luminosity
+                lumCount += 1
+
+    lumAverage = lumTotal / lumCount #(windowWidth * windowHeight)
+    print('checkpoint1')
+    print('average luminosity: ')
+    print lumAverage
+    print lumCount
+
+    for x in range(0, windowWidth - 1):
+        for y in range(0, windowHeight - 1):
+            currentRed = window.get_at((x, y)).r
+            currentGreen = window.get_at((x, y)).g
+            currentBlue = window.get_at((x, y)).b
+            currentAlpha = window.get_at((x, y)).a
+
+            luminosity = math.sqrt(math.pow(currentRed, 2) +
+                                   math.pow(currentGreen, 2) +
+                                   math.pow(currentBlue, 2))
+
+            if luminosity > lumAverage:
+                for i in range (0, tailLength):
+                    if x - i > 0:
+                        newRed =  currentRed - (currentRed * i / tailLength)
+                        newGreen = currentGreen - (currentGreen * i / tailLength)
+                        newBlue = currentBlue - (currentBlue * i / tailLength)
+
+                        pxArray[x - i, y] = (newBlue, newGreen, newRed)
+            else:
+                pxArray[x, y] = BLACK
+
+    print ('checkpoint2')
+
+    pict = pxArray.make_surface()
+
+    del pxArray
```

Comment:
You create and re-create the pixel array several times. Consider how you could refactor your code so you only need to do this once (each game loop).